src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py

```diff
@@ -479,23 +479,34 @@ def get_ou_id(self, ou_path, parent_ou_id=None):
             parent_ou_id = self.root_id
 
         # Parse ou_path and find the ID
-        ou_hierarchy = ou_path.strip("/").split("/")
-        hierarchy_index = 0
+        ou_path_as_list = ou_path.strip('/').split('/')
 
-        while hierarchy_index < len(ou_hierarchy):
+        for ou in ou_path_as_list:
             org_units = self.list_organizational_units_for_parent(parent_ou_id)
-            for ou in org_units:
-                if ou["Name"] == ou_hierarchy[hierarchy_index]:
-                    parent_ou_id = ou["Id"]
-                    hierarchy_index += 1
+
+            for org_unit in org_units:
+                if org_unit["Name"] == ou:
+                    parent_ou_id = org_unit["Id"]
                     break
             else:
-                raise ValueError(
-                    f"Could not find ou with name {ou_hierarchy} in OU list {org_units}.",
-                )
+                LOGGER.info(f'No OU found with name {ou} and parent {parent_ou_id}, will create.')
```

Comment:
Use lazy % formatting in logging functions. See details here: https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

And in the linter:

```sh
============================== 17 passed in 0.30s ==============================
# Linter performs static analysis to catch latent bugs
find ./src -iname "*.py" | xargs pylint --verbose --rcfile .pylintrc
Using config file .pylintrc
************* Module bootstrap_repository.adf-build.shared.python.organizations
src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py:492:16: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py:506:12: W0707: Consider explicitly re-raising using 'except Exception as exc' and 'raise OrganizationsException() from exc' (raise-missing-from)
```

The same file has other `LOGGER` examples that use lazy % formatting.

You can also run the linter locally to catch these errors. Review: ============================== 17 passed in 0.30s ==============================
# Linter performs static analysis to catch latent bugs
find ./src -iname "*.py" | xargs pylint --verbose --rcfile .pylintrc
Using config file .pylintrc
************* Module bootstrap_repository.adf-build.shared.python.organizations
src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py:492:16: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py:506:12: W0707: Consider explicitly re-raising using 'except Exception as exc' and 'raise OrganizationsException() from exc' (raise-missing-from)

Reply:
Thanks for this. Have been able to run this locally now and seems to be passing.

Unsure about the implementation about the exception which I've asked @StewartW for input on.